apps/system/js/base_module.js

```diff
@@ -121,33 +189,50 @@
         return;
       }
 
+      var submodules = this.constructor.SUB_MODULES.slice();
+      var unloaded = [];
+      submodules.forEach(function(submodule) {
+        if (BaseModule.defined(submodule) || window[submodule]) {
+          var name = BaseModule.lowerCapital(submodule);
+          if (!this[name]) {
+            this._initialSubModule(name, submodule);
+          }
+        } else {
+          unloaded.push(submodule);
+        }
+      }, this);
+
+      if (unloaded.length === 0) {
+        this.__sub_module_loaded && this.__sub_module_loaded();
```

Comment:
Is the `__` prefix supposed to be for "stuff the child class should/can override"?
I think we need to be more consistent with those (we also have `_start`, `_stop` etc...).
We even have a comment mentioning `onSubModuleInited` ??

How about:
```
baseStart()
baseStop()
baseSubModuleLoaded()
baseSideModuleLoaded()
```
etc...